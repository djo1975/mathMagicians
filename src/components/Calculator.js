import { useState } from 'react';
import calculate from '../logic/calculate';

const obj = {
  total: null,
  next: null,
  operation: null,
};

export default function Calculator() {
  const [output, setOutput] = useState(obj);
  const [activeBtn, setActiveBtn] = useState(null);

  const gumbs = ['AC', '+/-', '%', 'รท', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', '='];
  const handleBtnClick = (e) => {
    const btn = e.target.name;
    setOutput(calculate(output, btn));
    setActiveBtn(btn);
  };

  const screen = (item) => {
    if ((!item.total) && (!item.next)) {
      return '0';
    }
    if (!item.total) {
      return item.next;
    }
    if (item.next && item.total && item.operation) {
      return item.next;
    }
    return item.total;
  };

  return (
    <section id="calculator">
      <input
        type="text"
        value={screen(output)}
        disabled
        id="exit"
      />
      <section id="input">
        <article id="digits">
          {gumbs.map((gumb) => (
            <button
              key={gumb}
              type="button"
              name={gumb}
              className={activeBtn === gumb ? 'active' : ''}
              onClick={handleBtnClick}
            >
              {gumb}
            </button>
          ))}
        </article>
      </section>
    </section>
  );
}
