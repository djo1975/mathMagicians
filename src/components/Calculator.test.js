import React from 'react';
import { create } from 'react-test-renderer';
import Calculator from './Calculator';

describe('Calculator', () => {
  test('renders correctly', () => {
    const calculator = create(<Calculator />);
    expect(calculator.toJSON()).toMatchSnapshot();
  });
});
