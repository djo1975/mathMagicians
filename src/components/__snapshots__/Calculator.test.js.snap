// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div
  className="component-app"
>
  <div
    className="calculator-body"
  >
    <div
      className="component-display"
    >
      <div>
        0
      </div>
    </div>
    <div
      className="component-button-panel"
    >
      <div>
        <button
          className="component-button"
          onClick={[Function]}
        >
          AC
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          +/-
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          %
        </button>
        <button
          className="component-button orange"
          onClick={[Function]}
        >
          รท
        </button>
      </div>
      <div>
        <button
          className="component-button"
          onClick={[Function]}
        >
          7
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          8
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          9
        </button>
        <button
          className="component-button orange"
          onClick={[Function]}
        >
          x
        </button>
      </div>
      <div>
        <button
          className="component-button"
          onClick={[Function]}
        >
          4
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          5
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          6
        </button>
        <button
          className="component-button orange"
          onClick={[Function]}
        >
          -
        </button>
      </div>
      <div>
        <button
          className="component-button"
          onClick={[Function]}
        >
          1
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          2
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          3
        </button>
        <button
          className="component-button orange"
          onClick={[Function]}
        >
          +
        </button>
      </div>
      <div>
        <button
          className="component-button  wide"
          onClick={[Function]}
        >
          0
        </button>
        <button
          className="component-button"
          onClick={[Function]}
        >
          .
        </button>
        <button
          className="component-button orange"
          onClick={[Function]}
        >
          =
        </button>
      </div>
    </div>
  </div>
  <div />
</div>
`;

exports[`Calculator should render correctly 1`] = `ShallowWrapper {}`;

exports[`Calculator should renders correctly 1`] = `ShallowWrapper {}`;
